cmd/alertmanager/main.go

```diff
@@ -117,7 +117,13 @@ func main() {
 	}
 
 	logger := log.NewLogger(os.Stderr)
-	mrouter := initMesh(*meshListen, *hwaddr, *nickname, *password)
+
+	var mrouter *mesh.Router
+	if *meshListen == "" {
+		mrouter = initMesh(*flag.String("flag-name", "0.0.0.0:0", "flag-msg"), *hwaddr, *nickname, *password)
```

Comment:
I assume this is to provide a dummy mesh router because things below would fail if it is nil?

So right now silence and nflog actually attempt to gossip their data to a dummy mesh router, that is not actually running. I'm not sure there are any guarantees on all methods against an un-started router returning etc.

As both can be configured to not gossip their data, I think we should rather do that by not providing the Gossip/Mesh config options to them. That seems generally safer/cleaner.